src/Esprima/Utils/ToJavascriptConverter.cs

```diff
@@ -0,0 +1,1370 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Esprima.Ast;
+
+namespace Esprima.Utils
+{
+    public static class ToJavascriptConverterExtension

```

Comment:
Make it internal for now and add `ToJavaScriptString()` on `Program`.
The idea is that we parse, get an instance of `Program` (the parse tree) and can call `ToJavaScriptString()` on it directly, without having to deal with an alternate class.
Making it internal will let us change it later without breaking changes.

Reply:
If I do so, I need to copy the whole class to my Code, cause I use it to minify ES6 Template Literals with HTML & CSS inside, see:

    public class BaseCustomWebcomponentMinifier : ToJavascriptConverter
    {
        public static string ToJavascript(Node node)
        {
            var visitor = new BaseCustomWebcomponentMinifier();
            visitor.Visit(node);
            return visitor.ToString();
        }

        protected override void VisitPropertyDefinition(PropertyDefinition propertyDefinition)
        {
            if (TryGetParentAt(1) is ClassBody && propertyDefinition.Static && propertyDefinition.Key is Identifier i)
            {
                if (i.Name == "style")
                {
                    if (propertyDefinition.Value is TaggedTemplateExpression tte)
                    {
                        if (tte.Tag is Identifier iT && iT.Name=="css")
                        {
                            if (tte.Quasi is TemplateLiteral tl)
                            {
                                if (tl.Quasis.Count==1 && tl.Quasis[0] is TemplateElement)
                                {
                                    var q = tl.Quasis[0];
                                    var css = q.Value.Raw;
                                    try
                                    {
                                        var parser = new CssParser();
                                        var ss = parser.ParseStyleSheet(css);
                                        var newCss = ss.ToCss(new AngleSharp.Css.MinifyStyleFormatter());
                                        css = newCss;
                                    }
                                    catch (Exception ex)
                                    { }

                                    _sb.Append("static style=css`" + css + "`;");
                                    return;
                                }
                            }
                        }
                    }
                }
                if (i.Name == "template")
                {
                    if (propertyDefinition.Value is TaggedTemplateExpression tte)
                    {
                        if (tte.Tag is Identifier iT && iT.Name == "html")
                        {
                            if (tte.Quasi is TemplateLiteral tl)
                            {
                                if (tl.Quasis.Count == 1 && tl.Quasis[0] is TemplateElement)
                                {
                                    var q = tl.Quasis[0];
                                    var html = q.Value.Raw;
                                    try
                                    {
                                        var parser = new HtmlParser();
                                        var h = parser.ParseDocument(html);
                                        var newHtml = h.ToHtml(new AngleSharp.Html.MinifyMarkupFormatter());
                                        html = newHtml;
                                    }
                                    catch (Exception ex)
                                    { }

                                    _sb.Append("static template=html`" + html + "`;");
                                    return;
                                }
                            }
                        }
                    }
                }
            }
            base.VisitPropertyDefinition(propertyDefinition);
        }
    }

cause of that I also wanted to convert the Visitor to a real Visitor wich can return a modified AST